packages/playwright-test/src/worker/testInfo.ts

```diff
@@ -257,6 +261,8 @@ export class TestInfoImpl implements TestInfo {
           error,
         };
         this._onStepEnd(payload);
+        const errorForTrace = error ? { name: '', message: error.message || '', stack: error.stack } : undefined;
+        this._traceEvents.push(createAfterActionTraceEventForStep(stepId, serializeAttachments(this.attachments, initialAttachments), errorForTrace));
```

Comment:
This will pick up the same attachment for both steps:

```js
await test.step('outer', async () => {
  await test.step('inner', async () => {
    await test.info().attach(...);
  });
});
```

Reply:
Let's start like that, this way we can render 'attachment' in the report against the step that has attachments.